pom.xml

```diff
@@ -45,11 +55,52 @@
                 <groupId>net.alchim31.maven</groupId>
                 <artifactId>scala-maven-plugin</artifactId>
                 <version>${scala-maven-plugin.version}</version>
+                <!-- Common scala compiler configuration -->
+                <configuration>
+                    <scalaVersion>${scala-library.version}</scalaVersion>
+                </configuration>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>compile</goal>
+                            <goal>testCompile</goal>
+                        </goals>
+                    </execution>
+                </executions>
             </plugin>
         </plugins>
 
         <pluginManagement>
             <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <version>${maven-surefire-plugin.version}</version>
+                    <!-- Common unit test configuration -->
+                    <configuration>
+                        <parallel>methods</parallel>
+                        <redirectTestOutputToFile>true</redirectTestOutputToFile>
+                        <threadCount>10</threadCount>
+                    </configuration>
+                </plugin>
+
+                <plugin>
+                    <groupId>org.scalatest</groupId>
+                    <artifactId>scalatest-maven-plugin</artifactId>
+                    <version>${scalatest-maven-plugin.version}</version>
+                    <configuration>
+                        <reportsDirectory>${project.build.directory}/surefire-reports</reportsDirectory>
+                        <junitxml>.</junitxml>
```

Comment:
That's just a reporting mechanism. We can take it out if it bothers you.

http://www.scalatest.org/user_guide/using_the_scalatest_maven_plugin